WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,160 @@ Private.event_prototypes = {
+            currencyInfo.accountQuantity = currencyInfo.quantity
+            currencyInfo.realAccountQuantity = currencyInfo.quantity
+            if currencyInfo.transferPercentage then
+                currencyInfo.realCharacterQuantity = math.floor(currencyInfo.quantity * (currencyInfo.transferPercentage/100))
+            end
+
+            accountCurrencyData = Private.ExecEnv.FetchCurrencyDataFromAccountCharacters(
+              currencyInfo.currencyID,
+              event == "CURRENCY_TRANSFER_LOG_UPDATE"
+            ) or {}
+
+            for _, currencyData in ipairs(accountCurrencyData) do
+              if currencyData.quantity then
+                local realQuantity = currencyData.quantity
+                if currencyInfo.transferPercentage and currencyInfo.transferPercentage > 0 and currencyInfo.transferPercentage < 100 then
+                  realQuantity = math.floor(realQuantity * (currencyInfo.transferPercentage/100))
+                end
+                currencyData.realCharacterQuantity = realQuantity
+                currencyInfo.realAccountQuantity = currencyInfo.realAccountQuantity + realQuantity
+                currencyInfo.accountQuantity = currencyInfo.accountQuantity + currencyData.quantity
+              end
+            end
+          else
+            C_CurrencyInfo.RequestCurrencyDataForAccountCharacters()
+          end
+        end
+
+        local active = true
+        local primaryCheckFailed = false
+
+        -- Check quantity-related values
+        for attribute, triggerValue in pairs(triggerQuantities) do
+          if triggerValue ~= "nil" and not check(currencyInfo[attribute] or 0, triggerValue, triggerOperators[attribute]) then
+            active = false
+            if attribute ~= "quantity" and attribute ~= "realCharacterQuantity" then
+              primaryCheckFailed = true
+              break;
+            end
+          end
+        end
+
+        -- Helper function to check tristate values
+        local function checkTristate(tristateValue, attribute, condition, primary)
+          if active and tristateValue ~= "nil" and condition then
+            if tostring(attribute) ~= tristateValue then
+              active = false
+              primaryCheckFailed = primary and true
+            end
+          end
+        end
+
+        -- Check tristate values
+        checkTristate(discovered, currencyInfo.discovered, true, true)
+        checkTristate(capped, currencyInfo.quantity >= (currencyInfo.maxQuantity or 0), currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, false)
+        checkTristate(seasonCapped, currencyInfo.totalEarned >= (currencyInfo.maxQuantity or 0), currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, true)
+        checkTristate(weeklyCapped, currencyInfo.quantityEarnedThisWeek >= (currencyInfo.maxWeeklyQuantity or 0), currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0, true)
```

Comment:
Same as above, the checking whether to include the check for discovered/capped/seasonCapped/weeklyCapped should probably be done at function creation time.